controllers/chat/testFCMToken.js

```diff
@@ -0,0 +1,38 @@
+const {messaging} = require('firebase-admin');
+const {ResponseSuccess} = require('../../utils/Responses');
+
+const checkFCMToken = async (userToken) => {
+  const payload = {
+    notification: {
+      title: 'Test Notification',
+      body: 'Test'
+    },
+    token: userToken
+  };
+
+  try {
+    await messaging().send(payload);
+    console.log('Token is valid.');
+    return true; // Token is valid
+  } catch (error) {
+    if (error.code === 'messaging/registration-token-not-registered') {
+      console.error(`Token ${userToken} is invalid: ${error.message}`);
+      return false; // Token is invalid
+    } else {
+      console.error(`Error sending message: ${error.message}`);
+      return false; // Error occurred, but we treat token as invalid
+    }
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `checkFCMToken` logs errors directly to the console. It would be better to use a logging library or service that can provide more context and control over log levels, formats, and destinations. This will help in debugging and maintaining the application.

```diff
+ const logger = require('../../utils/logger'); // Assuming you have a logger utility

  try {
    await messaging().send(payload);
-   console.log('Token is valid.');
+   logger.info('Token is valid.');
    return true; // Token is valid
  } catch (error) {
    if (error.code === 'messaging/registration-token-not-registered') {
-     console.error(`Token ${userToken} is invalid: ${error.message}`);
+     logger.error(`Token ${userToken} is invalid: ${error.message}`);
      return false; // Token is invalid
    } else {
-     console.error(`Error sending message: ${error.message}`);
+     logger.error(`Error sending message: ${error.message}`);
      return false; // Error occurred, but we treat token as invalid
    }
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->